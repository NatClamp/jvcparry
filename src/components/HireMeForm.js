import React, { useState, useEffect } from 'react';
import {
  Div, Input, Button, Text, Icon, Anchor,
} from 'atomize';
import { useForm } from 'react-hook-form';
import axios from 'axios';

const HireMeForm = () => {
  const {
    register, handleSubmit, reset, formState: { isSubmitSuccessful },
  } = useForm();
  const [error, setError] = useState(false);
  const [submittedData, setSubmittedData] = useState({});
  const [thanksMessage, setThanksMessage] = useState(false);
  const [submitLoading, setSubmitLoading] = useState(false);

  const onSubmit = (data) => {
    axios.post(`${process.env.REACT_APP_JVCPARRY_API}/send`, data)
      .then((result) => {
        setSubmittedData(result);
        setSubmitLoading(false);
        setThanksMessage(true);
      })
      .catch((err) => {
        setSubmitLoading(false);
        console.log('Message not sent due to error');
        console.log(err);
        setError(err);
      });
  };

  useEffect(() => {
    if (isSubmitSuccessful) {
      setSubmitLoading(true);
      reset({ ...submittedData });
    }
  }, [isSubmitSuccessful, submittedData, reset]);

  return (
    <>
      <Div d="flex" flexDir="column" w={{ xs: '100%', lg: '100%' }} justify="center" className="HireMeFormContainer">
        <form onSubmit={handleSubmit(onSubmit)} w="100%">
          <Text tag="label" htmlFor="name" textSize="subheader">Name</Text>
          <Input
            {...register('name', { required: true })}
            type="text"
            m={{ y: '1rem' }}
          />
          <Text tag="label" htmlFor="email" textSize="subheader">Email</Text>
          <Input
            {...register('email', { required: true })}
            type="email"
            m={{ y: '1rem' }}
          />
          <Text tag="label" htmlFor="message" textSize="subheader">Your Message</Text>
          <Input
            tag="textarea"
            {...register('message')}
            m={{ y: '1rem' }}
            h={{ xs: '3rem', sm: '5rem' }}
            type="text"
          />

          <Div d="flex">
            <Button
              suffix={(
                <Icon
                  name="LongRight"
                  size="16px"
                  color="white"
                  m={{ l: '1rem' }}
                />
              )}
              shadow="3"
              hoverBg="hsla(217, 14%, 43%)"
              bg="hsla(217, 14%, 50%)"
              hoverShadow="4"
              type="submit"
              m={{ r: '1rem' }}
            >
              Contact Us
            </Button>
            {error && (
            <Text tag="p" textSize="body" textColor="danger800" p={{ t: '10px' }}>
              Sorry, an error occurred. Please contact me directly on
              {' '}
              <Anchor textColor="danger800" hoverTextColor="danger900" textWeight="600" href="mailto:jvcparry@hotmail.com">jvcparry@hotmail.com</Anchor>
            </Text>
            )}
            {submitLoading && <Text tag="p" textSize="body" textColor="black300" p={{ t: '10px' }}>Sending...</Text>}
            {thanksMessage && <Text tag="p" textSize="body" textColor="black300" p={{ t: '10px' }}>Thanks for your mail! I'll be in touch shortly.</Text>}
          </Div>
        </form>
      </Div>
    </>
  );
};

export default HireMeForm;
