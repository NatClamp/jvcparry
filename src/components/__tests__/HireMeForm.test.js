/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom/extend-expect';

import HireMeForm from '../HireMeForm';

describe('HireMeForm component', () => {
  const history = createMemoryHistory();
  afterEach(cleanup);

  it('should render the HireMeForm component', () => {
    render(
      <Router history={history}>
        <HireMeForm />
      </Router>,
    );
    expect(screen.getByText(/Contact Us/i)).toBeInTheDocument();
  });
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <Router history={history}>
          <HireMeForm />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
