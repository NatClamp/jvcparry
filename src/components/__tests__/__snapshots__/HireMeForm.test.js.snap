// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HireMeForm component should match the snapshot 1`] = `
<div
  className="HireMeFormContainer "
>
  <form
    onSubmit={[Function]}
    w="100%"
  >
    <label
      className=""
      htmlFor="name"
    >
      Name
    </label>
    <div
      className=""
    >
      <input
        className=""
        d="flex"
        m={
          Object {
            "y": "1rem",
          }
        }
        name="name"
        type="text"
      />
    </div>
    <label
      className=""
      htmlFor="email"
    >
      Email
    </label>
    <div
      className=""
    >
      <input
        className=""
        d="flex"
        m={
          Object {
            "y": "1rem",
          }
        }
        name="email"
        type="email"
      />
    </div>
    <label
      className=""
      htmlFor="message"
    >
      Your Message
    </label>
    <div
      className=""
    >
      <textarea
        className=""
        d="flex"
        m={
          Object {
            "y": "1rem",
          }
        }
        name="message"
        type="text"
      />
    </div>
    <div
      className=""
    >
      <button
        className=""
        type="submit"
      >
        Contact Us
        <svg
          className=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5857864,13 L6,13 C5.44771525,13 5,12.5522847 5,12 C5,11.4477153 5.44771525,11 6,11 L15.5857864,11 L12.2928932,7.70710678 C11.9023689,7.31658249 11.9023689,6.68341751 12.2928932,6.29289322 C12.6834175,5.90236893 13.3165825,5.90236893 13.7071068,6.29289322 L18.7071068,11.2928932 C19.0976311,11.6834175 19.0976311,12.3165825 18.7071068,12.7071068 L13.7071068,17.7071068 C13.3165825,18.0976311 12.6834175,18.0976311 12.2928932,17.7071068 C11.9023689,17.3165825 11.9023689,16.6834175 12.2928932,16.2928932 L15.5857864,13 Z"
            fill="#FFFFFF"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
  </form>
</div>
`;
